packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -2164,8 +2169,12 @@ class _CupertinoTimerPickerState extends State<CupertinoTimerPicker> {
   }
 
   Widget _buildHourPicker(EdgeInsetsDirectional additionalPadding, Widget selectionOverlay) {
+    final FixedExtentScrollController scrollController = FixedExtentScrollController(
+        initialItem: selectedHour!
+    );
```

Comment:
The controllers should not be created in every build. There should be just one. How about creating field _scrollController and initializing it in initState?

Reply:
I changed the approach to 3 distinct `FixedExtentScrollController` variables. My concern is that this slightly changes the behavior. While I see no apparent problems myself, someone more experienced with the flutter source might want to have a look at this.

Reply:
Its valid concern. @HansMuller , can you take a look please?

Reply:
Not sure if anyone depends on the current behavior although running a complete tap_presubmit would be helpful in this case. 

Presumably, to restore the original behavior we'd need update the scrollControllers at build time with `jumpToItem`

Reply:
How can i kick tap_presubmit for a PR?

Reply:
@NobodyForNothing , I got understanding how to run tap presubmit for the change.

Can you apply `jumpToItem` inside the build to the controller and then I will run testing and merge?

Reply:
@polina-c 
I am not quite sure that I get what is meant by "apply `jumpToItem`". 
The way I understand it `FixedExtentScrollController.jumpToItem` is used to change which item of the scroll view is centered. I can't find any references to its connection to the constructor in the documentation and code.

Could you just run tap presubmit to see whether the current replacement is enough? In case this PR changes relevant behavior I assume it is covered by tests.

Reply:
Testing does not guarantee the change is right 😀 We need both testing and agreement that code modifications make sense. 

Let me explain about `jumpToItem`.

In original code we created controller in build and asked it to set initial selection to current, for example `selectedSecond`.
Now we create controller in the beginning and reuse it. And, selectedSecond may change. So, we need the controller to move to new selection at build time. 

Does it help?
Thank you for pushing this forward in spite of difficulties!!!

Reply:
Thanks for the clarification, I completely forgot about the `initialItem` parameter. I changed the behavior accordingly.